sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -239,13 +239,46 @@ class pi_shared_mutex : public std::shared_mutex {
   }
 };
 
+// This wrapper around std::atomic is created to limit operations with reference
+// counter and to make allowed operations more transparent in terms of
+// thread-safety in the plugin.
```

Comment:
Please explain in comment that these operations do not need a mutex guard around them since the underlying data is already atomic (plus extra semantics that is explained for "decrement_and_count")

Reply:
Added comment.